docs/Writing_easyconfig_files.rst

```diff
@@ -180,7 +186,162 @@ of the form ``('<checksum value>', '<checksum type>')``.
+
+    $ diff -u /example/bzip2-1.0.6.eb.bak_20170824200906 /example/bzip2-1.0.6.eb
+    diff --git a//example/bzip2-1.0.6.eb.bak_20170824200906 b/example/bzip2-1.0.6.eb
+    index 46b2debed..2eb73f15a 100644
+    --- a/example/bzip2-1.0.6.eb.bak_20170824200906
+    +++ b/example/bzip2-1.0.6.eb
+    @@ -9,8 +9,9 @@ compressors), whilst being around twice as fast at compression and six times fas
+     toolchain = {'name': 'dummy', 'version': 'dummy'}
+     toolchainopts = {'pic': True}
+
+    -sources = [SOURCE_TAR_GZ]
+     source_urls = ['http://www.bzip.org/%(version)s/']
+    +sources = [SOURCE_TAR_GZ]
+    +checksums = ['a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd']
+
+     buildopts = "CC=gcc CFLAGS='-Wall -Winline -O3 -fPIC -g $(BIGFILES)'"
+
+.. note:: Along with injecting checksums, EasyBuild will also reorder the ``source_urls``, ``sources``
+          and ``patches`` specifications, in that order and if they are present, and include the ``checksums``
+          specification afterwards. This is done to facilitate working towards a uniform style in easyconfig files,
+          which also applies to the order of specified easyconfig parameters.
+
+.. _inject_checksums_replacing:
+
+Replacing existing checksums
+****************************
+
+When one or more checksums are already specified, EasyBuild requires the use of ``--force`` together
+with ``--inject-checksums`` to replace those checksums. A clear warning will be printed to notify
+that exsiting checksums will be replaced.
+
+For example::
+
+    $ eb bzip2-1.0.6.eb --inject-checksums
+    == temporary log file in case of crash /tmp/eb-WhSwVH/easybuild-HCODnl.log
+    == injecting sha256 checksums in /example/bzip2-1.0.6.eb
+    == fetching sources & patches for bzip2-1.0.6.eb...
+    ERROR: Found existing checksums, use --force to overwrite them
+
+.. code:: bash
+
+    $ eb bzip2-1.0.6.eb --inject-checksums --force
+    == temporary log file in case of crash /tmp/eb-dS2QLa/easybuild-JGxOzC.log
+    == injecting sha256 checksums in /example/bzip2-1.0.6.eb
+    == fetching sources & patches for bzip2-1.0.6.eb...
+
+    WARNING: Found existing checksums in bzip2-1.0.6.eb, overwriting them (due to use of --force)...
+
+    == backup of easyconfig file saved to /example/bzip2-1.0.6.eb.bak_20170824203850...
+    == injecting sha256 checksums for sources & patches in bzip2-1.0.6.eb...
+    ...
+
+.. note::
+    Any existing checksums are *blindly* replaced when ``--inject-checksums --force`` is used:
+    the existing checksumsthey are *not verified* to be correct as during normal use of EasyBuild
+    (since that would kind of defeat the purpose of ``--inject-checksums``).
+
+    In addition, it also doesn't matter whether or not checksums are available for all sources & patches:
+    with ``--inject-checksums``, checksums will be added for *all* sources and patches,
+    including for extensions listed in ``exst_list`` (if any).
```

Comment:
`exts_list`...now I'm starting to doubt myself

Reply:
No, you're right, I was just making that typo consistently... ;-)

```
$ eb -a | grep exts_list
exts_list           List with extensions added to the base installation [default: []]